steward/src/cork.rs

```diff
@@ -94,6 +108,121 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
+        let request = request.get_ref();
+
+        let contract_call_data = match request.call_data.clone() {
+            Some(call) => call,
+            None => {
+                return Err(tonic::Status::invalid_argument(
+                    "Error, can't find call data",
+                ))
+            }
+        };
+
+        let config = APP.config();
+
+        let name = &config.keys.delegate_key;
+        let wallet = config.load_ethers_wallet(name.to_string());
+
+        let eth_host = config.ethereum.rpc.clone();
+        let provider = Provider::<Http>::try_from(eth_host.clone())
+            .unwrap_or_else(|_| panic!("could not get provider from eth_host {}", eth_host));
+        let chain_id = provider
+            .get_chainid()
+            .await
+            .expect("Could not retrieve chain ID");
+
+        let chain_id =
+            downcast_to_u64(chain_id).expect("Chain ID overflowed when downcasting to u64");
+        let client = SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id));
+
+        let cellar_address = request.cellar_id.clone();
+
+        if cellars::validate_cellar_id(&cellar_address).is_err() {
+            return Err(tonic::Status::invalid_argument(
+                "Error, can't validate Cellar ID",
+            ));
+        }
+
+        let address: Address = cellar_address
+            .parse()
+            .expect("could not parse cellar address");
+
+        match contract_call_data {
+            AaveV2Stablecoin(call) => {
+                let contract = AaveV2StablecoinCellar::new(address, Arc::new(client));
+
+                match call.function.unwrap() {
+                    Function::EnterPosition(_) => contract_call(contract.enter_position()).await,
+                    Function::SetDepositLimit(params) => {
+                        contract_call(contract.set_deposit_limit(params.limit.into())).await
+                    }
+                    Function::Reinvest(params) => {
+                        contract_call(contract.reinvest(params.min_assets_out.into())).await
+                    }
+                    Function::Rebalance(params) => {
+                        contract_call(
+                            contract.rebalance(
+                                params
+                                    .route
+                                    .iter()
+                                    .map(|addr| match addr.parse::<H160>() {
+                                        Ok(addr) => Ok(addr),
+                                        Err(_) => Err(addr),
+                                    })
+                                    .collect::<Vec<_>>()
+                                    .iter()
+                                    .map(|r| r.unwrap())
+                                    .collect::<Vec<H160>>()
+                                    .try_into()
+                                    .expect("failed to convert 'route' addresses to array"),
```

Comment:
Trying to do this all as one long statement directly as an argument to `rebalance` makes this a bit difficult to read, and also means you can't call `validate_route` on it like the other version is doing. It would be cleaner to define some intermediary variables to break up this long chain similar to how it's parsed in the existing code.

If we find that we are repeating a bunch of this code, perhaps it would be best to split it into a function in a single place that both can call.